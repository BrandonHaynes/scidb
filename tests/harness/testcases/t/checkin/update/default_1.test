--setup
--start-query-logging
create temp array A <a:int32 default 10> [x=0:2,3,0]

--test
--aql select * into A from build(A,x+9)
--aql update A set a = a + 1
--aql select * into A from filter(build(A,x+12),x<>1)
--aql update A set a = a + 1
--aql update A set a = a + 20
--aql update A set a = a - 21
--aql update A set a = 10
--aql select a from A
--aql select a + 0 from A
--aql update A set a = a * 1
--aql update A set a = a / 1
--aql update A set a = a * 10
--aql update A set a = a / 10
--aql update A set a = a / 2
--aql select * into A from build(A,x+9)

--cleanup
remove(A)
--stop-query-logging
