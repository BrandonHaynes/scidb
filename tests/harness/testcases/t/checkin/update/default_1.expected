SCIDB QUERY : <create temp array A <a:int32 default 10> [x=0:2,3,0]>
Query was executed successfully

SCIDB QUERY : <select * into A from build(A,x+9)>
{x} a
{0} 9
{1} 10
{2} 11

SCIDB QUERY : <update A set a = a + 1>
{x} a
{0} 10
{1} 11
{2} 12

SCIDB QUERY : <select * into A from filter(build(A,x+12),x<>1)>
{x} a
{0} 12
{2} 14

SCIDB QUERY : <update A set a = a + 1>
{x} a
{0} 13
{2} 15

SCIDB QUERY : <update A set a = a + 20>
{x} a
{0} 33
{2} 35

SCIDB QUERY : <update A set a = a - 21>
{x} a
{0} 12
{2} 14

SCIDB QUERY : <update A set a = 10>
{x} a
{0} 10
{2} 10

SCIDB QUERY : <select a from A>
{x} a
{0} 10
{2} 10

SCIDB QUERY : <select a + 0 from A>
{x} expr
{0} 10
{2} 10

SCIDB QUERY : <update A set a = a * 1>
{x} a
{0} 10
{2} 10

SCIDB QUERY : <update A set a = a / 1>
{x} a
{0} 10
{2} 10

SCIDB QUERY : <update A set a = a * 10>
{x} a
{0} 100
{2} 100

SCIDB QUERY : <update A set a = a / 10>
{x} a
{0} 10
{2} 10

SCIDB QUERY : <update A set a = a / 2>
{x} a
{0} 5
{2} 5

SCIDB QUERY : <select * into A from build(A,x+9)>
{x} a
{0} 9
{1} 10
{2} 11

SCIDB QUERY : <remove(A)>
Query was executed successfully

